packages/camera/camera_avfoundation/example/ios/RunnerTests/FLTCamPhotoCaptureTests.m

```diff
@@ -94,4 +94,86 @@ - (void)testCaptureToFile_mustReportPathToResultIfSavePhotoDelegateCompletionsWi
   [self waitForExpectationsWithTimeout:1 handler:nil];
 }
 
+- (void)testCaptureToFile_mustSetExtensionToHeicIfAvailablePhotoCodecTypesContainsHEVC {
+  XCTestExpectation *expectation =
+      [self expectationWithDescription:
+                @"Test must set extension to heif if availablePhotoCodecTypes contains HEVC."];
+  dispatch_queue_t captureSessionQueue = dispatch_queue_create("capture_session_queue", NULL);
+  dispatch_queue_set_specific(captureSessionQueue, FLTCaptureSessionQueueSpecific,
+                              (void *)FLTCaptureSessionQueueSpecific, NULL);
+  FLTCam *cam = FLTCreateCamWithCaptureSessionQueue(captureSessionQueue);
+  [cam setFileFormat:FCPFileFormatHEIF];
+
+  AVCapturePhotoSettings *settings =
+      [AVCapturePhotoSettings photoSettingsWithFormat:@{AVVideoCodecKey : AVVideoCodecTypeHEVC}];
+
+  id mockSettings = OCMClassMock([AVCapturePhotoSettings class]);
+  OCMStub([mockSettings photoSettingsWithFormat:OCMOCK_ANY]).andReturn(settings);
+
+  NSString *filePath = @"test";
+  id mockResult = OCMClassMock([FLTThreadSafeFlutterResult class]);
+  OCMStub([mockResult sendSuccessWithData:filePath]).andDo(^(NSInvocation *invocation) {
+    [expectation fulfill];
+  });
+
+  id mockOutput = OCMClassMock([AVCapturePhotoOutput class]);
+  // Set availablePhotoCodecTypes to HEVC
+  NSArray *codecTypes = @[ AVVideoCodecTypeHEVC ];
+  OCMStub([mockOutput availablePhotoCodecTypes]).andReturn(codecTypes);
+
+  OCMStub([mockOutput capturePhotoWithSettings:OCMOCK_ANY delegate:OCMOCK_ANY])
+      .andDo(^(NSInvocation *invocation) {
+        FLTSavePhotoDelegate *delegate = cam.inProgressSavePhotoDelegates[@(settings.uniqueID)];
+        // Completion runs on IO queue.
+        dispatch_queue_t ioQueue = dispatch_queue_create("io_queue", NULL);
```

Comment:
Labeling it "io_queue" doesn't make it the same as the actual IO queue of FLTCam. Can you either get the queue from FLTCam, or inject this IO queue into FLTCam?

Reply:
Sorry never mind - the actual FLTCam's implementation that relies on IO queue is mocked out. So any queue here is fine.